parsl/tests/test_python_apps/test_fail.py

```diff
@@ -1,89 +1,62 @@
-import argparse
-
-import parsl
 from parsl.app.app import python_app
-from parsl.tests.configs.local_threads import fresh_config as local_config
+from parsl.dataflow.errors import DependencyError
 
 
-@python_app
-def sleep_fail(sleep_dur, sleep_rand_max, fail_prob, inputs=[]):
-    import time
-    import random
+class ManufacturedTestFailure(Exception):
+    pass
 
-    s = sleep_dur + random.randint(-sleep_rand_max, sleep_rand_max)
 
-    time.sleep(s)
-    raise Exception("App failure")
+@python_app
+def random_fail(fail_prob, inputs=()):
+    import random
+    if random.random() < fail_prob:
+        raise ManufacturedTestFailure("App failure")
 
 
-def test_no_deps(numtasks=2):
+def test_no_deps(ntasks=3):
     """Test basic error handling, with no dependent failures
     """
+    futs = [random_fail(.8) for i in range(ntasks)]
```

Comment:
I don't see why this test needs to be randomised at all: this test seems to be hoping that some app invocations will fail in this range (otherwise, no failure handling is exercise) and so now I question why this should be done on a random basis rather than a deterministic list of (for example) two successful apps and one non-successful one.

Reply:
Och, good point.  I think I was trying to match the intent of the original `random`.  But not seeing that I did a good job.  All that needs be done is removing the `.sleep()` call.